packages/api/src/services/profileValidation.js

```diff
@@ -20,7 +21,10 @@ function profileValidation(user) {
   }
   if (user.role.toLowerCase() === 'student') {
     if (!Reflect.has(user, 'batchId') || user.batchId.length < 24) {
-      return { passed: false, msg: ERR_MSGS.noBatchId };
+      const batchIdInDB = Batch.findOne({ _id: user.batchId });
```

Comment:
There should be an await here, otherwise batchIdInDB is just a Promise object which is always truthy.